### PR TITLE
Polish the CPU profiler status view

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR replaces the linear progress indicator with a circular progress indicator, and adds different processing information for the phases `fetching` and `processing`. This gives the user a better idea of what is happening instead of seeing a linear progress indicator that is stuck at zero for a long time.

This PR also makes sure we are showing the status view when work is happening (like filtering by user tag for example).

cpu-processing-info
There is still a little bit of a hang we may be able to improve using batch processing or isolates, but this is an improvement over the hang that existed previously. This .gif was taken in debug mode, too, so release perf will be better.

RELEASE_NOTE_EXCEPTION=too small to mention

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler_controller.dart [hunk 1]
```diff
@@ -387,82 +397,85 @@ class CpuProfilerController extends DisposableController
...
-    // Look up the stored app start up profiles before calling [reset]. This
-    // will save us the work of processing the startup profiles again if we have
-    // already processed them once.
-    var appStartUpProfile = await lookupProfileAndProcess(
-      label: appStartUpUserTag,
-    );
-    final appStartUpProfileWithFilters = await lookupProfileAndProcess(
-      label: _wrapWithFilterTag(appStartUpUserTag),
-    );
+    Future<void> loadAppStartUpProfileHelper() async {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profiler_controller.dart:400]: review this with hide white space enabled

File: packages/devtools_app/lib/src/screens/profiler/panes/controls/cpu_profiler_controls.dart [hunk 2]
```diff
@@ -71,31 +71,6 @@ Example queries:
   }
 }
 
-class CpuProfilerDisabled extends StatelessWidget {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/panes/controls/cpu_profiler_controls.dart:74]: moved to file profile_status.dart

File: packages/devtools_app/lib/src/screens/profiler/panes/controls/profiler_screen_controls.dart [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+import 'package:flutter/material.dart';
+
+import '../../../../shared/analytics/constants.dart' as gac;
+import '../../../../shared/common_widgets.dart';
+import '../../../../shared/globals.dart';
+import '../../../../shared/theme.dart';
+import '../../../../shared/ui/vm_flag_widgets.dart';
+import '../../profiler_screen_controller.dart';
+
+class ProfilerScreenControls extends StatelessWidget {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/panes/controls/profiler_screen_controls.dart:14]: all these controls were moved from profiler_screen.dart

File: packages/devtools_app/lib/src/screens/profiler/profiler_screen.dart [hunk 4]
```diff
@@ -139,243 +126,50 @@ class _ProfilerScreenBodyState extends State<ProfilerScreenBody>
...
-          ),
-        );
-        final emptyProfileView = Center(
-          child: RichText(
-            textAlign: TextAlign.center,
-            text: const TextSpan(
-              text: 'No CPU samples recorded.',
-            ),
-          ),
-        );
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/profiler_screen.dart:167]: these were made into stateless widgets and moved to profiler_status.dart

File: packages/devtools_app/lib/src/screens/profiler/profiler_status.dart [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+class ProfilerBusyStatus extends StatelessWidget {
+  const ProfilerBusyStatus({super.key, required this.status});
+
+  final CpuProfilerBusyStatus status;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(status != CpuProfilerBusyStatus.none);
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/profiler/profiler_status.dart:110]: Nit: can this be asserted in the constructor instead?

**Review by bkonyi — APPROVED** [review]

**Pull request merged**